client/src/security/keyprovider.rs

```diff
@@ -35,6 +38,237 @@ impl TryFrom<Vec<u8>> for KeyProvider {
     }
 }
 
+/// Constructor functions for KeyProvider
+impl KeyProvider {
+    /// Creates a new [`KeyProvider`] with a `passphrase` of arbitrary length.
+    ///
+    /// The `passphrase` will be truncated to 32 bytes of length internally. This function
+    /// is being provided to support legacy passphrase unlock of the snapshot-file.
+    ///
+    /// # Example
+    /// ```
+    /// use iota_stronghold::KeyProvider;
+    /// use std::ops::Deref;
+    ///
+    /// let passphrase = b"superlongpassphraseextrasecure";
+    /// let mut expected: [u8; 32] = [0u8; 32];
+    /// passphrase
+    ///     .as_ref()
+    ///     .iter()
+    ///     .enumerate()
+    ///     .take(32)
+    ///     .for_each(|(i, value)| {
+    ///         expected[i] = *value;
+    ///     });
```

Comment:
copy_from ?

Reply:
unfortunately does not handle under or overflow :'(